### ISSUE TITLE
SIGSEGV of target process on Ubuntu 14.04.2 LTS 64bit when attaching to any process

### ISSUE THREAD

**Issue by isegal** [frida/frida#75]
Version 5.0.10 of Frida causes the target process to segfault immediately after attachment.

Steps to reproduce:

- Following the Functions hello.c tutorial from http://www.frida.re/docs/functions/

- As soon as hook.py is run, hello.c crashes with SIGSEGV.

This also happens with any other process and with any frida tool that attaches, such as frida-trace.

```
Program terminated with signal SIGSEGV, Segmentation fault.
#0  0x00007f55df65c912 in _dl_lookup_symbol_x (undef_name=0x7f55df86f440 "frida_agent_main", undef_map=0x0, ref=0x7f55df06dee8, symbol_scope=0x388,
    version=0x0, type_class=0, flags=2, skip_map=0x0) at dl-lookup.c:731
731     dl-lookup.c: No such file or directory.
(gdb) t apply all bt

Thread 2 (Thread 0x7f55df866740 (LWP 5612)):
#0  0x00007f55df34df20 in __nanosleep_nocancel () at ../sysdeps/unix/syscall-template.S:81
#1  0x00007f55df34ddd4 in __sleep (seconds=0) at ../sysdeps/unix/sysv/linux/sleep.c:137
#2  0x00000000004005db in main ()

Thread 1 (Thread 0x7f55df06e700 (LWP 5627)):
#0  0x00007f55df65c912 in _dl_lookup_symbol_x (undef_name=0x7f55df86f440 "frida_agent_main", undef_map=0x0, ref=0x7f55df06dee8, symbol_scope=0x388,
    version=0x0, type_class=0, flags=2, skip_map=0x0) at dl-lookup.c:731
#1  0x00007f55df3c3145 in do_dlsym (ptr=ptr@entry=0x7f55df06ded0) at dl-libc.c:97
#2  0x00007f55df661ff4 in _dl_catch_error (objname=0x7f55df06deb0, errstring=0x7f55df06dec0, mallocedp=0x7f55df06dea0, operate=0x7f55df3c3100 <do_dlsym>,
    args=0x7f55df06ded0) at dl-error.c:187
#3  0x00007f55df3c3249 in dlerror_run (args=0x7f55df06ded0, operate=0x7f55df3c3100 <do_dlsym>) at dl-libc.c:46
#4  __GI___libc_dlsym (map=<optimized out>, name=<optimized out>) at dl-libc.c:210
#5  0x00007f55df86f10a in ?? ()
#6  0x0000000000000000 in ?? ()
```

glibc version:
```
GNU C Library (Ubuntu EGLIBC 2.19-0ubuntu6.6) stable release version 2.19, by Roland McGrath et al.
Copyright (C) 2014 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.
There is NO warranty; not even for MERCHANTABILITY or FITNESS FOR A
PARTICULAR PURPOSE.
Compiled by GNU CC version 4.8.2.
Compiled on a Linux 3.13.11 system on 2015-02-25.
Available extensions:
        crypt add-on version 2.1 by Michael Glad and others
        GNU Libidn by Simon Josefsson
        Native POSIX Threads Library by Ulrich Drepper et al
        BIND-8.2.3-T5B
libc ABIs: UNIQUE IFUNC
```

Please let me know if you need anything else to help troubleshoot.

**Comment by oleavr** [comment]
Thanks for reporting! I have heard of similar reports earlier, will try to investigate this soon.

**Comment by oleavr** [comment]
Sorry about the delay. This is now finally [fixed](https://github.com/frida/frida/commit/aef24365643bdd303f527e69f6fa97baac3e9d83) and released in 6.0.8.
